cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -843,6 +851,65 @@ impl ManagedEnvironment {
         Ok(result)
     }
 
+    /// Create a new generation from local changes,
+    /// and updates the generation lock.
+    ///
+    /// If no changes exist, returns without further action.
+    ///
+    /// TODO: this should do a `build` before committing to a generation
+    pub fn sync_local(&self, flox: &Flox) -> Result<(), ManagedEnvironmentError> {
+        let mut local = self.local_checkout(flox)?;
+
+        if Self::validate_checkout(&local, &self.get_current_generation(flox)?) {
+            debug!("local checkout and remote checkout equal, nothing to apply");
+            return Ok(());
+        }
+
+        let mut generations = self
+            .generations()
+            .writable(flox.temp_dir.clone())
+            .map_err(ManagedEnvironmentError::CreateFloxmetaDir)?;
+
+        generations
+            .add_generation(
+                &mut local,
+                "Synchronized manual changes to generation".to_string(),
+            )
+            .unwrap();
+        self.lock_pointer()?;
+        Ok(())
+    }
+
+    /// Create or reset a local checkout of the latet generation.
+    ///
+    /// Copies the `env/` directory from the latest generation to the `.flox/` directory
+    /// and returns a [CoreEnvironment] for the `.flox/env`.
+    fn local_checkout(&self, flox: &Flox) -> Result<CoreEnvironment, ManagedEnvironmentError> {
+        if !self.path.join(ENV_DIR_NAME).exists() {
+            debug!("creating environment directory");
+            let latest_gen = self.get_current_generation(flox)?;
+            fs::create_dir_all(self.path.join(ENV_DIR_NAME))
+                .map_err(ManagedEnvironmentError::CreateLocalEnvironmentView)?;
+            copy_dir_recursive(&latest_gen.path(), &self.path.join(ENV_DIR_NAME), true)
+                .map_err(ManagedEnvironmentError::CreateLocalEnvironmentView)?;
+        }
+
+        let local = CoreEnvironment::new(self.path.join(ENV_DIR_NAME));
+        Ok(local)
+    }
+
+    fn validate_checkout(local: &CoreEnvironment, remote: &CoreEnvironment) -> bool {
+        let local_manifest =
+            toml::from_str::<toml::Value>(&fs::read_to_string(local.manifest_path()).unwrap());
+        let remote_manifest =
+            toml::from_str::<toml::Value>(&fs::read_to_string(remote.manifest_path()).unwrap());
+
+        // todo: validate lockfile and otehr files as well
+        // todo: require binary equal?
+
+        local_manifest == remote_manifest
+    }
```

Comment:
regarding these todos:
- do we want to keep it at the manifest for now, since except the lockfile we don't purposefully manage any other files yet and previous discussion suggested that we dont need to check the lockfile.

- i think we should look at binary equality of the manifest, since otherwise comments cant be added locally